docs/network_discovery.md

```diff
@@ -35,13 +35,13 @@ from homeassistant.components import bluetooth
 ...
 
 @callback
-def _async_discovered_device(service_info: bluetooth.BluetoothServiceInfo, change: bluetooth.BluetoothChange) -> None:
+def _async_discovered_device(service_info: bluetooth.BluetoothServiceInfoBleak, change: bluetooth.BluetoothChange) -> None:
```

Comment:
```suggestion
def _async_discovered_device(service_info: bluetooth.BluetoothServiceInfoBleak, change: bluetooth.BluetoothChange) -> None:
```
I would rather keep the original naming for the interface, if there is no very clear reason not to do so. The bleak may not exist forever while `BluetoothServiceInfo` might.

Reply:
Active Integrations frequently need to access `.device` since they have to connect to the device.  If we don't type it to the Bleak one, it is not obvious its available.

Reply:
Okay, fine by me, it just felt wrong to me to have an implementation specific name for something that looks like an interface :-)